Readme.md

```diff
@@ -0,0 +1,1432 @@
+```
+
+This feature affects type inference across parameters list and in the case of case classes it excludes all members of the non-first parameter lists from the `equals`, `hashCode` and `unapply` implementations.
+
+The latter point especially very often causes code which relies on multiple parameter lists to be confusing. Therefore, using them in general is forbidden.
+
+There are two use cases for which multiple parameters lists are allowed. The first one is definition of implicit arguments (see the respective section above):
+
+```scala
+def doSomeWork(arg: String)(implicit ec: ExecutionContext) = ...
+```
+
+The second one is when the second parameter list is used to accept a single functional value:
+
+```scala
+def run(x: Int)(f: Int => String): String = f(x)
+
+run(10) { x => x.toString }
+```
+
+Multiple parameter lists here allow using more concise syntax for passing an anonymous function which increases readability. That said, the need to declare such functions in itself should be rare, because of the rule against DSLs.
+
+The second use case, however, does not extend to multiple parameter lists which accept functions; in other words, something like the following is not allowed:
+
+```scala
+def computeAnswer(onComplete: Int => Unit)
+                 (onFailure: Throwable => Unit) = ...
+
+computeAnswer { x => println(x) } { e => e.printStackTrace() }
+```
+
+If you do need to create a function which, for example, accepts multiple callbacks, use named parameters instead:
+
+```scala
+def computeAnswer(onComplete: Int => Unit,
+                  onFailure: Throwable => Unit) = ...
+
+run(
+  onComplete = { x => println(x) },
+  onFailure = { e => e.printStackTrace }
+)
+```
+
+## Symbolic methods and infix and postfix notation
+
+Scala allows to define methods which consist of symbols instead of regular letters. Combined with  the ability to call any method with the infix notation this allows one to define custom operators:
+
+```scala
+class Connectable {
+  def =+~>(other: Connectable): Connectable = ...
+}
+
+val x = new Connectable
+val y = new Connectable
+x =+~> y
+```
+
+Symbolic method names do not convey what they do, and with the exception of regular well-known symbols (mathematical operations, `++`/`--` on collections, boolean operators, negation, etc.) tend to be obscure by themselves: `a =+~> x` could mean absolutely anything.
+
+Therefore it is prohibited to define custom symbolic methods, while their use with the 3rd-party libraries should be as limited as possible. Naturally, using unicode characters, which is also allowed by the language, strictly prohibited as well.
+
+Additionally, using the infix notation for calling regular, non-symbolic methods is discouraged and could be used only if it significantly increases readability. One example of a sufficiently motivated use of infix notation are matchers and other parts of the tests definition in ScalaTest.
```

Comment:
This is true. Infix operators came from mathematics, and I can tell you for sure that there are virtually no infix operators which are not related to some kind of mathematical structure and therefore assume some kind of symmetry of the arguments, at least, in the most fundamental and central areas.

Moreover, if we take an overview of most of the programming language with the conventional, ALGOL-based syntax, we will find that almost always operators there are used for fundamentally symmetric things, like those I've described above. Of course, for languages which allow operator overriding, this is less true, but even in them it is almost always suggested not to override these operators for things which they are not intended for. I would say that among all well-known languages with C heritage, Scala (and Kotlin) are the only ones which allow using regular, non-symbolic methods as infix operators.

As for the official Scala guide, it is quite understandable - why they would advise against a feature they have introduced? In my opinion, the reasons stated above and in the guide are sufficient to prohibit using infix syntax in most cases, and I'm not the only one - [Databricks guide](https://github.com/databricks/scala-style-guide#infix-methods) also suggests avoiding them.